airflow/www/views.py

```diff
@@ -510,6 +511,7 @@ def dag_stats(self):
         return wwwutils.json_response(payload)
 
     @expose('/task_stats')
+    @login_required
```

Comment:
Is there a reason you're adding @login_required to these stat endpoints? It would make it more difficult if we tried to integrate monitoring tools like [Prometheus](https://github.com/prometheus/prometheus) etc. to read from these endpoints if we need to make it authenticate. If it's just giving information on dag stats and task stats, I think it's fine to not require authentication.

Reply:
If you secure airflow, it shouldn't expose any information by default. Even the dagids could be sensitive. For prometheus you could use the experimental api + http basic authentication. That's what we are using.